### DISCUSSION TITLE
How to do auth on web app based on Cloudflare Workers

### DISCUSSION THREAD

**Discussion by rishavs** [supabase/supabase#9443] [Questions]
Hey folks

I am making a web app on Cloudflare Workers, with Supabase providing the data/auth layer and I wanted to understand how to do server side auth with Google social login.

Here's my current understanding from reading through all the docs & sample apps;

1. Add supabase-js in the clidentside HTML via a <script> tag
2. Check for client side session using `supabase.auth.getSession()`
3. if session exists, get user details using `supabase.auth.getUser()`
4. if session doesnt exists, use supabase.auth.signIn to sign in into Google
6. In the clientside, use ``supabase.auth.getUser()` to get the access_token under currentSession and add to a httpOnly cookie
7. in the server side, again add supabse-js as a dependency
8. any time user tries to access a protected resource, parse the cookie, get the access_token and then use `supabase.auth.api.getUser(jwt)` to get the user details, and if that user is allowed to access the resource, let them.

Is this overall flow correct?
Where does the refreshing of tokens factors in here?
I am assuming supabase automatically handles the sessions. or Do I need to do anything here as well?
Can I skip all the client side auth steps altogether and have fully server side auth (with some cookie data saved on the client)?

**Comment by rishavs** [comment]
Pinging back on this. Can someone help?

**Comment by rishavs** [comment]
I am surprised that there doesn't seems to be anyway of doing this. Even more worried that noone is interested in answering these rather basic questions. I am not sure if I would be getting any help if I do decide to stick to Supabase for my webapp.

**Reply by steve-chavez** [reply]
I suggest asking on [Discord](https://discord.supabase.com), it's better for more open-ended questions like this.

**Reply by rishavs** [reply]
I actually came from Discord as most of my questions there had no answers. I was hoping that the Github discussion board might be better tracked.

**Reply by steve-chavez** [reply]
We do track github discussions but the team can't know every integration with third party tools. Why not use [Supabase edge functions](https://supabase.com/docs/guides/functions)? You might get more help that way. Perhaps a solution with edge functions would be transferrable to cloudflare workers?

**Reply by yw662** [reply]
> can't know every integration with third party tools. 

You can still answer the general part, eg. how to work with supabase auth from backend, or more generally how to work with supabase from backend.

**Comment by yw662** [comment]
There is little major difference between doing auth with cf worker/function as the backend and doing auth with a nodejs backend. So it is no more than "how to do auth with custom backend" and "how to work with supabase auth with a custom backend".

I have some ideas but it is merely "do the auth by yourself and see what supabase may provide". It is indeed not "integration with third party tools". It is a general question about using supabase from backend.
